nixos/modules/security/acme/default.nix

```diff
@@ -257,6 +292,12 @@ let
       # minica will output to a folder sharing the name of the first domain
       # in the list, which will be ${data.domain}
       script = ''
+        ${optionalString (!isNull(lockfileName)) ''
+          # wrap whole script in a subshell, lock releases after subshell exit
+          (
```

Comment:
There's a simplified version that works without the "double-ended" wrapping and without spawning a sub-shell:

https://stackoverflow.com/questions/60894739/how-do-exec-and-flock-work-together-in-bash-script

Reply:
Locks similarly arcane, but in the end appears to be cleaner indeed, thanks. Integrated this.